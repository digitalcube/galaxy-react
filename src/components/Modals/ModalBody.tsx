import React, { FC, PropsWithChildren } from 'react';

export type ModalBodyProps = PropsWithChildren<{
  onSubmit?: React.FormEventHandler<HTMLFormElement>;
}>;

export const ModalBody: FC<ModalBodyProps> = ({ children, onSubmit }) => {
  const className = 'modal-body px-4 py-4';
  if (onSubmit) {
    return (
      <form className={className} onSubmit={onSubmit}>
        {children}
      </form>
    );
  }
  return <div className={className}>{children}</div>;
};
