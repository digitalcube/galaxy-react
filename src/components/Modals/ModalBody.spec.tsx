import { render } from '@testing-library/react';
import React from 'react';
import { ModalBody } from './ModalBody';

describe('ModalBody', () => {
  it('to match snapshot [no submit handler]', () => {
    const { asFragment } = render(<ModalBody>Content</ModalBody>);
    expect(asFragment()).toMatchSnapshot();
  });
  it('to match snapshot [with submit event]', () => {
    const { asFragment } = render(
      <ModalBody onSubmit={(e) => e.preventDefault()}>Title</ModalBody>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
