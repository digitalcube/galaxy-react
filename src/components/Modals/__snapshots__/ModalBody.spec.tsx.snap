// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalBody to match snapshot [no submit handler] 1`] = `
<DocumentFragment>
  <div
    class="modal-body px-4 py-4"
  >
    Content
  </div>
</DocumentFragment>
`;

exports[`ModalBody to match snapshot [with submit event] 1`] = `
<DocumentFragment>
  <form
    class="modal-body px-4 py-4"
  >
    Title
  </form>
</DocumentFragment>
`;
